packages/forms/src/elements/common/Label.tsx

```diff
@@ -37,6 +38,26 @@ export const Label = React.forwardRef<HTMLLabelElement, ILabelProps>((props, ref
 
   if (fieldContext) {
     combinedProps = fieldContext.getLabelProps(combinedProps);
+
+    if (type === undefined) {
+      const { setIsLabelActive, setIsLabelHovered } = fieldContext;
+
+      combinedProps = {
+        ...combinedProps,
+        onMouseUp: composeEventHandlers(props.onMouseUp, () => {
+          setIsLabelActive(false);
+        }),
+        onMouseDown: composeEventHandlers(props.onMouseDown, () => {
+          setIsLabelActive(true);
+        }),
```

Comment:
By controlling focus from the `mouseDown` callback, I'm not sure this matches native functionality as expected.

If someone were to `mouseDown` the label, but then drag away and not fully click, this would still trigger the minimum thumb to be selected.

From our offline discussion, it may be easier for the `MinThumb` to provide its `ref` via context. This `Label` could then have something like:

```
onClick: () => minThumbRef.current.focus()
```

Reply:
> If someone were to mouseDown the label, but then drag away and not fully click, this would still trigger the minimum thumb to be selected.

After some pair programming with @austin94 and some exploration, I've concluded that the only way to achieve such an interaction is leveraging an event listener on the `document` that listens for a `mouseup` event:

```jsx
useEffect(() => {
  if (fieldContext) {
    if (isLabelActive && minThumbRef.current) {
      minThumbRef.current.focus();
    }

    const listener = () => {
      if (!fieldContext.isLabelHovered) {
        minThumbRef.current && minThumbRef.current.blur();
        setIsLabelActive && setIsLabelActive(false);
        document.body.focus();
      }
    };

    window.document.documentElement.addEventListener('mouseup', listener);

    return () => {
      window.document.documentElement.removeEventListener('mouseup', listener);
    };
  }

  return undefined;
  }, [fieldContext, isLabelActive, setIsLabelActive]);
```

We've concluded that the above code is invasive due to the global event listener and decided to keep the interaction as-is.

Reply:
[nit] something about this interaction seems to be preventing `user-select` on the label. In other words, I can't drag to select label text for copy like I can with the native control.

Reply:
Awesome, thanks for the catch. I'll take a look at this issue.

Reply:
It looks like when the `mousedown` event is fired (and the min thumb is simultaneously focused) text selection is interrupted. 

I've decoupled the min thumb focus from the `mousedown` event by focusing on the min thumb on a `click` event instead which fixes the text selection issue.

Reply:
It looks like when the `mousedown` event is fired (and the min thumb is simultaneously focused) text selection is interrupted. 

I've decoupled the min thumb focus from the `mousedown` event by focusing on the min thumb on a `click` event instead which fixes the text selection issue.

Edit: @jzempel I've re-requested a review for the [new commit](https://github.com/zendeskgarden/react-components/pull/812/commits/8caf0eeb64667d05b3b44669db3830624751abff) that addresses the text selection issue.